### PR TITLE
TEST/uct_hello_world: added -pedantic flag to check uct.h in examples

### LINKED ISSUE #1270: ucp.h and uct.h should compile with -Wpedantic
```
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h: In function ‘ucs_cpu_set_find_lcs’:
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h:58:5: warning: ISO C forbids braced-groups within expressions [-Wpedantic]
01:25:10      ({ \
01:25:10      ^
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h:71:13: note: in expansion of macro ‘UCS_CPU_ISSET’
01:25:10          if (UCS_CPU_ISSET(i, cpu_mask)) {
01:25:10              ^
01:25:10 In file included from /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucp/api/ucp.h:13:0,
01:25:10                  from pml_ucx.h:21,
01:25:10                  from pml_ucx.c:13:
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h: In function ‘ucs_cpu_set_find_lcs’:
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h:58:5: warning: ISO C forbids braced-groups within expressions [-Wpedantic]
01:25:10      ({ \
01:25:10      ^
01:25:10 /var/lib/jenkins/jobs/gh-ompi-master-pr/workspace/ucx_local/install/include/ucs/type/cpu_set.h:71:13: note: in expansion of macro ‘UCS_CPU_ISSET’
01:25:10          if (UCS_CPU_ISSET(i, cpu_mask)) {
01:25:10              ^
``
solution using “__extension__” specifier:
https://github.com/pmix/master/blob/master/src/dstore/pmix_esh.c#L71

### PR DESCRIPTION
**Pull request by shssf**
This PR should fix #1270

### CODE REVIEW

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -716,7 +716,7 @@ struct ucp_tag_recv_info {
  * the run-time environment. Then, the fetched descriptor is used for
  * UCP library @ref ucp_init "initialization". The Application can print out the
  * descriptor using @ref ucp_config_print "print" routine. In addition
- * the application is responsible to @ref ucp_config_free "free" the
+ * the application is responsible to @ref ucp_config_release "release" the
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: pls revert this

File: src/uct/api/uct.h [hunk 2]
```diff
@@ -145,74 +145,76 @@ typedef struct uct_tl_resource_desc {
  *
  * The enumeration list presents a full list of operations and capabilities
  * exposed by UCX API.
+ * @{
```
Comment (yosefe) [inline src/uct/api/uct.h]: pls change 'The enumeration...'

File: src/uct/api/uct.h [hunk 3]
```diff
@@ -145,74 +145,76 @@ typedef struct uct_tl_resource_desc {
...
+ */
+        /* Active message capabilities */
+#define UCT_IFACE_FLAG_AM_SHORT       UCS_BIT(0)  /**< Short active message */
```
Comment (yosefe) [inline src/uct/api/uct.h]: can you please check if ucx.pdf is generated correctly after this change?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1043/ for details.

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (shssf) [inline src/ucp/api/ucp.h]: This is a typo - we have no `ucp_config_free` function

Comment (yosefe) [inline src/ucp/api/ucp.h]: right, but it should be in another PR ("UCP/API: Documentation fixes")

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1044/ for details.

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3002/ for details (Mellanox internal link).

File: src/uct/api/uct.h (hunk 3 above)
Comment (shssf) [inline src/uct/api/uct.h]: Doxygen assumes macro definitions as a functions in topic definition. I added one more subgroup and it looks nice at the moment. Also, this subgroup allow to set group reference.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1047/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3003/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp pls take a look

**Comment by shamisp** [comment]
👍 @bbenton - please take a look.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1052/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1053/ for details.

**Comment by bbenton** [comment]
:+1: from doc/API  perspective

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/1057/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3007/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/3011/ for details (Mellanox internal link).
